### PR TITLE
Support bytebuddy as proxy

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Motivation:

Explain the context, and why you're making that change.
To make others understand what is the problem you're trying to solve.

### Modification:

this issue https://github.com/alipay/sofa-rpc/issues/489
Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>. 

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

File: core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxy.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                .make()
+                .load(interfaceClass.getClassLoader(), ClassLoadingStrategy.Default.INJECTION)
+                .getLoaded();
+
+            PROXY_CLASS_MAP.put(interfaceClass, cls);
+        }
+        try {
+            return cls.newInstance();
+        } catch (Throwable t) {
+            throw new SofaRpcRuntimeException("", t);
```
Comment (JervyShi) [inline core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxy.java]: Add some message for this exception.

File: core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxyTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    public void getProxy() throws Exception {
+        BytebuddyProxy proxy = new BytebuddyProxy();
+        AbstractTestClass testClass = null;
+        try {
+            testClass = proxy.getProxy(AbstractTestClass.class, new TestInvoker());
+        } catch (Exception e) {
+            LOGGER.info(e.getMessage());
+        }
+        Assert.assertNotNull(testClass);
```
Comment (JervyShi) [inline core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxyTest.java]: Wrong interface class still can get proxy class?

Comment (leizhiyuan) [reply core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxyTest.java]: yes, bytebuddy support abstract class. sofa will prevent this.but as an proxy,we do not need to.

File: core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxy.java (hunk 1 above)
Comment (leizhiyuan) [reply core-impl/proxy/src/main/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxy.java]: fixed

File: core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxyTest.java (hunk 2 above)
Comment (JervyShi) [reply core-impl/proxy/src/test/java/com/alipay/sofa/rpc/proxy/bytebuddy/BytebuddyProxyTest.java]: OK

**Comment by khotyn** [comment]
Any documentation?

**Comment by lazyfighter** [comment]
english bytebuddy doc:[http://bytebuddy.net/#/](url)
chinese bytebuddy doc:[https://notes.diguage.com/byte-buddy-tutorial/](url)

**Pull request merged**